parser.go

```diff
@@ -479,22 +479,26 @@ func (p *Parser) parseSelectItem() ast.SelectItem {
 	}
 }
 
-func (p *Parser) tryParseAsAlias() *ast.AsAlias {
+func (p *Parser) tryParseAsAlias(requiredAs bool) *ast.AsAlias {
```

Comment:
I feel like `p.tryParseAsAlias( /* requiredAs */ false)` is a little difficult to write.
Can't we introduce two method for `requiredAs=true` and `requiredAs=False`? (like (`tryParseAsAliasAsRequired` and `tryParseAsAliasAsOptional`)

Reply:
`tryParseAsAliasAsRequired` seems like a strange name to someone unfamiliar with this project (e.g. twice `As` looks like typo, and `try` and `required` seem contradictory), and it is hard to guess what the method does. By making the explicit argument, it is clear that the method parses `AS name` as possible, and the comment makes it clear that the argument gives the required `AS`.

Reply:
My thought:
> twice As looks like typo and try and required seem contradictory
I feel that the current `AsAlias` can be `Alias`, and the current `Alias` is like `ExprWithAlias` or `AliasedExpr`.
`tryParseAliasAsRequired`, and `tryParseAliasAsOptional` is felt better.

> the comment makes it clear that the argument gives the required AS.

I think maintaining `/* requiredAs */` comment is harder because it can't be assisted by go-lsp or IDEs, and it is strange to someone(me) unfamiliar with this project. At least, it must be noted in doc comment of `tryParseAsAlias()`.

Reply:
> `tryParseAliasAsRequired` also seems strange to me because `As` looks like a usual preposition, and `try` and `required` have conflicting meanings.

It means it can be better if the subject of "required" or "optional" is unambiguous.
The subject is "keyword `AS`", so they can be called as `tryParseAsAliasRequireKeyword`, `tryParseAsAliasOptionalKeyword`.

Reply:
It's ambiguous enough that there are multiple methods to parse a particular struct, and I don't think it's a good idea to distinguish them by method names. If it really should be parsed by multiple methods, we should introduce multiple structs, but I don't think in this case.

Reply:
> It's ambiguous enough that there are multiple methods to parse a particular struct

I agree structs and methods should be one by one.
In the other language, it may be a usecase of missing named parameter.
In Go, it may be better to introduce const with enum semantics.

Reply:
Okay, because of code jumps and editor popups, I decided to remove the `/* requiredAs */` comment.

Reply:
We have agreed about `tryParseAsAlias()`, so it is just my thought.
> I think that `Alias` is the identifier after `AS`, not the `AS name` itself. 

SQL has common structure "alias, which is optionally prefixed by `AS`", and they are safe to unparse always with `AS`.
I prefer to call it `Alias` than `AsAlias` because someones have coding guideline which prefer to omit `AS`.
As far as I know, `AS` of table alias is often omitted. (Probably due to Oracle's past not allowing `AS`.)

Reply:
The name `Alias` ​​has less information than the name `AsAlias`, so I don't think it should be changed proactively.